assignments/lambda-classes.js

```diff
@@ -1 +1,182 @@
 // CODE here for your Lambda Classes
+
+class Person {
+  constructor(obj) {
+    this.name = obj.name;
+    this.age = obj.age;
+    this.location = obj.location;
+    this.gender = obj.gender;
+  }
+
+  speak() {
+    return `Hello my name is ${this.name}, I am from ${this.location}.`;
+  }
+}
+
+class Instructor extends Person {
+  constructor(obj) {
+    super(obj);
+    this.specialty = obj.specialty;
+    this.favLanguage = obj.favLanguage;
+    this.catchPhrase = obj.catchPhrase;
+  }
+
+  demo(subject) {
+    console.log(`Today we are learning about ${subject}`);
+  }
+
+  grade(student, subject) {
+    console.log(`${student.name} receives a perfect score on ${subject}`);
+  }
+
+  adjustGrade(student) {
+    student.grade = Math.floor(Math.random() * 100) + 1;
+  }
+}
+
+class ProjectManager extends Instructor {
+  constructor(obj) {
+    super(obj);
+    this.gradClassName = obj.gradClassName;
+    this.favInstructor = obj.favInstructor;
+  }
+
+  standUp(channel) {
+    console.log(`${this.name} announces to ${channel}, @channel standy times!`);
+  }
+
+  debugsCode(student, subject) {
+    console.log(`${this.name} debugs ${student.name}'s code on ${subject}`);
+  }
+}
+
+class Student extends Person {
+  constructor(obj) {
+    super(obj);
+    this.previousBackground = obj.previousBackground;
+    this.className = obj.className;
+    this.favSubjects = obj.favSubjects;
+    this.grade = obj.grade;
+  }
+
+  listsSubjects() {
+    const arr = this.favSubjects;
+    console.log(`${this.name}'s favorite subjects:`);
+    arr.forEach((x, y) => console.log(`  ${y + 1}. ${x}`));
+  }
+
+  PRAssignment(subject) {
+    console.log(`${this.name} has submitted a PR for ${subject}`);
+  }
+
+  sprintChallenge(subject) {
+    console.log(`${this.name} has begun sprint challenge on ${subject}`);
+  }
+
+  graduate() {
+    if (this.grade >= 70) {
+      console.log(`Congratulations ${this.name}! Your grade of ${this.grade} qualifies you for graduation.`);
+    } else {
+      const oldGrade = this.grade;
+      console.log(`${this.name} has a current grade of ${this.grade}%.`);
+      console.log(`Grading more of ${this.name}'s assignments...`);
+      jerry.adjustGrade(this);
```

Comment:
I really like what you did here, but it is a little awkward because we're relying on an instance of ProjectManager named jerry